geoportailv3/static/js/draw/drawnfeaturesservice.js

```diff
@@ -1,26 +1,203 @@
 /**
- * @fileoverview Provides a selected features service useful to share
+ * @fileoverview Provides a dranw features service useful to share
  * information about the selected feature throughout the application.
  */
 
 goog.provide('app.DrawnFeatures');
 
 goog.require('app');
+goog.require('ngeo.Location');
+goog.require('ngeo.format.FeatureHash');
 goog.require('ol.Collection');
 
 
+
+/**
+ * @constructor
+ * @param {ngeo.Location} ngeoLocation Location service.
+ * @param {app.Mymaps} appMymaps Mymaps service.
+ * @ngInject
+ */
+app.DrawnFeatures = function(ngeoLocation, appMymaps) {
+
+  /**
+   * @type {app.Mymaps}
+   * @private
+   */
+  this.appMymaps_ = appMymaps;
+
+  /**
+   * @type {ol.Collection}
+   */
+  this.features = new ol.Collection();
+
+  /**
+   * @type {ngeo.Location}
+   * @private
+   */
+  this.ngeoLocation_ = ngeoLocation;
+
+  /**
+   * @type {ngeo.format.FeatureHash}
+   * @private
+   */
+  this.fhFormat_ = new ngeo.format.FeatureHash({
+    encodeStyles: false,
+    properties: (
+        /**
+         * @param {ol.Feature} feature Feature.
+         * @return {Object.<string, (string|number)>} Properties to encode.
+         */
+        function(feature) {
+          // Do not encode the __editable__ and __selected__ properties.
+          var properties = feature.getProperties();
+          delete properties['__editable__'];
+          delete properties['__selected__'];
+          for (var key in properties) {
+            if (goog.isNull(properties[key])) {
+              delete properties[key];
+            }
+          }
+          return properties;
+        })
+  });
+
+  /**
+   * @type {ol.FeatureStyleFunction}
+   * @private
+   */
+  this.featureStyleFunction_ = this.appMymaps_.createStyleFunction();
+};
+
+
+/**
+ * @param {ol.Feature} feature
+ */
```

Comment:
I think you could add some doc to the function to explain what it does and when to use it.